src/GdbCommand.h

```diff
@@ -27,28 +28,146 @@ class GdbCommand {
   virtual std::string invoke(GdbServer& gdb_server, Task* t,
                              const std::vector<std::string>& args) = 0;
 
+  /**
+   * When call, gdb will automatically fill argument to this command
+   * by running gdb.execute(). This is useful to gdb state along side
+   * of the command invocation.
```

Comment:
Your first comment is pointer at auto_args. This feature is used to have gdb automatically populate argument. For instance getting the location of the current frame is much easier to do from gdb than it is from GdbServer since we don't have access to the symbolication features. In the checkpoint commands we're going to automatically call provider a 'where' argument invisible to the user.

The auto parse is unrelated to the auto_args. I think in insights it wasn't worth writing the auto parse code for it. But now that it all works I do think it's nice. It gives us a easy way to and consistent way to validate arguments and produce errors. If we go with parse_to_tuple it will really look like this for every command:
std::string error_msg;
if (auto args = parse_to_tuple<T1, T2, ...>(args, error_msg)) {
   ....
} else {
  return error_msg;
}
Botond originally wrote the code out for variable argument parsing but I decided to leave that out since it was pretty complicated. Reverting this means well lose the central validation logistic and we'll get to drop the DispatchTo code.

Do you still think we should drop this? I feel less strongly about this one.